### PR TITLE
DASD-4123 Documentation and keyvault ARM template

### PR DESCRIPTION
**Pull request by rivoric**
Added documentation, Test-ARMTemplate.ps1 and keyvault acceptance test

### CODE REVIEW

File: ArmTemplates/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# ARM Templates
+
+This folder contains ARM deployments that can be used to stand up a specific resource and can be used in the same way as nested templated.
```
Comment (NickGraham101) [inline ArmTemplates/README.md]: Aren't they linked templates if you're calling from a master file?  https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-linked-templates

File: Tests/sample.ps1 [hunk 2]
```diff
@@ -0,0 +1,18 @@
+$ResourceGroupName = "dfc-test-template-rg"
+$TemplateFile = "$PSScriptRoot\..\ArmTemplates\keyvault.json"
+$params = @{ keyVaultName = "dfc-foo-bar-kv" }
+
+$output = Test-AzureRmResourceGroupDeployment `
```
Comment (NickGraham101) [inline Tests/sample.ps1]: Should use a hashtable for readability and consistency

File: Tests/sample.ps1 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+$ResourceGroupName = "dfc-test-template-rg"
+$TemplateFile = "$PSScriptRoot\..\ArmTemplates\keyvault.json"
+$params = @{ keyVaultName = "dfc-foo-bar-kv" }
+
+$output = Test-AzureRmResourceGroupDeployment `
+              -ResourceGroupName $ResourceGroupName `
+              -TemplateFile $TemplateFile `
+              -TemplateParameterObject $params `
+              -ErrorAction Stop `
+              5>&1
```
Comment (NickGraham101) [inline Tests/sample.ps1]: Not sure what this does

Comment (rivoric) [reply Tests/sample.ps1]: sample.ps1 should have been deleted. I was using it for debugging - will fix.

File: ArmTemplates/README.md (hunk 1 above)
Comment (rivoric) [reply ArmTemplates/README.md]: That's exactly what they are. I had used the resource type - fixed and added link you provided

File: Tests/sample.ps1 (hunk 2 above)
Comment (rivoric) [reply Tests/sample.ps1]: Done

**Pull request merged**